Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Migration/Tools/MigrationTool.cs

```diff
@@ -0,0 +1,256 @@
+        /// </summary>
+        /// <param name="selectedObject">Unity object of type of a GameObject or SceneAsset</param>
+        public bool TryAddObjectForMigration(Object selectedObject)
+        {
+            if (!selectedObject)
+            {
+                Debug.LogError("Selection is Empty. Could not add for Migration.");
+                return false;
+            }
+
+            if (selectedObject is GameObject || selectedObject is SceneAsset)
+            {
+                if (!migrationObjects.Contains(selectedObject))
+                {
+                    migrationObjects.Add(selectedObject);
+                    Debug.Log($"{selectedObject.name} object added for Migration");
+                }
+                return true;
+            }
+            Debug.LogError("Object must be a GameObject, Prefab or SceneAsset. Could not add object for Migration");
+            return false;
+        }
+
+        /// <summary>
+        /// Adds all prefabs and scene assets found on the Assets folder to the list of objects to be migrated
+        /// </summary>
+        public void AddProjectForMigration()
+        {
+            AddAllAssetsOfTypeForMigration(new Type[] { typeof(GameObject), typeof(SceneAsset) });
+        }
+
+        private void AddAllAssetsOfTypeForMigration(Type[] types)
+        {
+            if (types == null)
+            {
+                return;
+            }
+            foreach (var type in types)
+            {
+                foreach (var path in FindAllAssetsOfType(type))
+                {
+                    TryAddObjectForMigration(AssetDatabase.LoadAssetAtPath(path, type));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Removes object from the list of objects to migrated
+        /// </summary>
+        /// <param name="selectedObject">Object to be removed</param>
+        public void RemoveObjectForMigration(Object selectedObject)
+        {
+            migrationObjects.Remove(selectedObject);
+        }
+
+        /// <summary>
+        /// Clears list of objects to be migrated
+        /// </summary>
+        public void ClearMigrationList()
+        {
+            migrationObjects.Clear();
+        }
+
+        /// <summary>
+        /// Migrates all objects from list of objects to be migrated using the selected IMigrationHandler implementation. 
+        /// </summary>
+        /// <param name="type">A type that implements IMigrationhandler</param>
+        /// <returns></returns>
+        public bool MigrateSelection(Type type, bool askToSaveCurrentScene)
+        {
+            if (!SetMigrationHandlerInstance(type))
+            {
+                return false;
+            }
+
+            if (askToSaveCurrentScene)
+            {
+                if (!EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo())
+                {
+                    Debug.Log("d");
```

Comment:
REmove?

Reply:
Should we let the user know at this point that we couldn't save the modified scenes? - else they might not know why we're aborting migration - maybe just change the log message to something that makes sense to the user